src/EventProducerService.h

```diff
@@ -33,13 +33,19 @@ class EventProducerService : public Service {
   void sendEvent(bool state, byte evValue, byte data1);
   void sendEvent(bool state, byte evValue, byte data1, byte data2);
   void sendEvent(bool state, byte evValue, byte data1, byte data2, byte data3);
+  void generateMessage();
+  void generateMessage(byte data1);
+  void generateMessage(byte data1, byte data2);
+  void generateMessage(byte data1, byte data2, byte data3);
 
 private:
   Controller *controller;
   Configuration *module_config;  // Shortcut to reduce indirection code.
   ConsumeOwnEventsService *coeService;
   void (*eventhandler)(byte index, VlcbMessage *msg);
 
+  byte opCode;
+  byte nn_en[4];
```

Comment:
Hmm, this doesn't look right. This data belongs to method calls and do not reflect the state of the EPService.
Yes, it is a technique used sometimes to save a few bytes, but I find it confusing.

Reply:
I did this to keep them in scope across the various methods that use them.  I can look again at using a lot of arguments in the various calls instead.